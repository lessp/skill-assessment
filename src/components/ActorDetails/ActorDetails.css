.App-actorDetails {
  padding: 2.4rem;
  position: relative; }
  .App-actorDetails h4 {
    font-weight: 600;
    font-size: 1.8rem; }
  .App-actorDetails header {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    padding-bottom: 1.2rem;
    margin-bottom: 1.2rem;
    border-bottom: 1px solid #f2f2f2; }
    .App-actorDetails header img {
      margin-right: 1.2rem;
      margin-bottom: 1.2rem;
      height: 25.6rem;
      width: auto;
      border-radius: 5px;
      display: block; }
    @media screen and (max-width: 420px) {
      .App-actorDetails header img {
        height: 19.2rem;
        margin-right: 0; } }
  .App-actorDetails section ul {
    display: flex;
    flex-direction: column;
    text-align: center; }
  @media screen and (max-width: 420px) {
    .App-actorDetails section h4 {
      text-align: center; } }
