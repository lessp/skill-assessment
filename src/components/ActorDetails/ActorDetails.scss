.App-actorDetails {
  padding: 2.4rem;
  margin: 2.4rem auto;

  width: calc(100% - 4.8rem);
  max-width: 96rem;

  background: white;
  border-radius: 5px;
  box-shadow: 0 12px 24px rgba(0, 0, 0, 0.25);

  position: relative;

  .closeButton {
    position: absolute;
    top: 1.2rem;
    left: 1.2rem;
    width: 3.2rem;
    height: 3.2rem;

    display: flex;
    justify-content: center;
    align-items: center;

    cursor: pointer;

    &::before,
    &::after {
      content: '';
      position: absolute;
      top: 48%;
      left: 30%;
      width: 40%;
      height: 0.2rem;
      background: var(--color-default);
      transform: rotate(45deg);
    }

    &::after {
      transform: rotate(-45deg);
    }
  }

  h4 {
    font-weight: 600;
    font-size: 1.8rem;
  }

  header {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;

    padding-bottom: 1.2rem;
    margin-bottom: 1.2rem;
    border-bottom: 1px solid #f2f2f2;

    img {
      margin-right: 1.2rem;
      margin-bottom: 1.2rem;
      height: 25.6rem;
      width: auto;
      border-radius: 5px;
      display: block;
    }

    @media screen and (max-width: 420px) {
      img {
        height: 19.2rem;
        margin-right: 0;
      }
    }
  }

  section {
    ul {
      display: flex;
      flex-direction: column;
      text-align: center;
    }

    @media screen and (max-width: 420px) {
      h4 {
        text-align: center;
      }
    }
  }
}
