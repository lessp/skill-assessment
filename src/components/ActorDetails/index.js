import React from 'react';

import './ActorDetails.css';

const ActorDetails = ({ actor, handleBackTransition }) => (
  <article className="App-actorDetails">
    <header>
      <img src={actor.details.profile_path} alt={actor.details.name} />
      <div>
        <h4>{actor.details.name}</h4>
        <p>
          <span className="strong">Born: </span>
          {actor.details.birthday} in {actor.details.place_of_birth}
        </p>
        {actor.details.deathday !== null && (
          <p>
            <span className="strong">Dead: </span>
            {actor.details.deathday}
          </p>
        )}
        <p>
          {actor.details.biography.length > 134
            ? actor.details.biography.substring(0, 134) + '...'
            : actor.details.biography}
        </p>
      </div>
    </header>
    <section>
      <ul>
        {actor.credits.slice(0, 10).map(movie => (
          <li key={movie.id}>
            <p>
              {movie.title} ({movie.character})
            </p>
          </li>
        ))}
      </ul>
    </section>
  </article>
);

export default ActorDetails;
