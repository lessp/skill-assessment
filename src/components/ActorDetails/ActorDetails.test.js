import React from 'react';
import ActorDetails from './';
import renderer from 'react-test-renderer';

import {
  actorCreditsUrl,
  actorDetailsUrl,
  normaliseActor,
  sortByPopularity
} from '../../utils';

it('renders ActorDetails with dummy API-data', () => {
  const actor = {
    details: {
      profile_path: 'dummyProfilePath.png',
      name: 'Dummy Actor Name',
      birthday: '1970-01-01',
      deathday: '2000-01-01',
      biography: 'Dummy biography.'
    },
    credits: [
      {
        id: '1',
        title: 'Dummy Movie Title',
        character: 'Dummy Character'
      }
    ]
  };

  const tree = renderer.create(<ActorDetails actor={actor} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ActorDetails with real API-data', async () => {
  const credits = await fetch(actorCreditsUrl('500')).then(res => res.json());
  const details = await fetch(actorDetailsUrl('500')).then(res => res.json());

  const tree = renderer
    .create(
      <ActorDetails
        actor={{
          details: normaliseActor(details),
          credits: sortByPopularity(credits.cast)
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
