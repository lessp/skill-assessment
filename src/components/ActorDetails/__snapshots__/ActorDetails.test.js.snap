// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ActorDetails with dummy API-data 1`] = `
<article
  className="App-actorDetails"
>
  <header>
    <img
      alt="Dummy Actor Name"
      src="dummyProfilePath.png"
    />
    <div>
      <h4>
        Dummy Actor Name
      </h4>
      <p>
        <span
          className="strong"
        >
          Born: 
        </span>
        1970-01-01
         in 
      </p>
      <p>
        <span
          className="strong"
        >
          Dead: 
        </span>
        2000-01-01
      </p>
      <p>
        Dummy biography.
      </p>
    </div>
  </header>
  <section>
    <ul>
      <li>
        <p>
          Dummy Movie Title
           (
          Dummy Character
          )
        </p>
      </li>
    </ul>
  </section>
</article>
`;

exports[`renders ActorDetails with real API-data 1`] = `
<article
  className="App-actorDetails"
>
  <header>
    <img
      alt="Tom Cruise"
      src="https://image.tmdb.org/t/p/w300//3oWEuo0e8Nx8JvkqYCDec2iMY6K.jpg"
    />
    <div>
      <h4>
        Tom Cruise
      </h4>
      <p>
        <span
          className="strong"
        >
          Born: 
        </span>
        1962-07-03
         in 
        Syracuse, New York, USA
      </p>
      <p>
        Thomas "Tom" Cruise (born Thomas Cruise Mapother IV; July 3, 1962) is an American actor and filmmaker. He has been nominated for three...
      </p>
    </div>
  </header>
  <section>
    <ul>
      <li>
        <p>
          Mission: Impossible - Rogue Nation
           (
          Ethan Hunt
          )
        </p>
      </li>
      <li>
        <p>
          Mission: Impossible
           (
          Ethan Hunt
          )
        </p>
      </li>
      <li>
        <p>
          Mission: Impossible - Ghost Protocol
           (
          Ethan Hunt
          )
        </p>
      </li>
      <li>
        <p>
          Interview with the Vampire
           (
          Lestat de Lioncourt
          )
        </p>
      </li>
      <li>
        <p>
          Mission: Impossible III
           (
          Ethan Hunt
          )
        </p>
      </li>
      <li>
        <p>
          Eyes Wide Shut
           (
          Dr. William 'Bill' Harford
          )
        </p>
      </li>
      <li>
        <p>
          Austin Powers in Goldmember
           (
          Tom Cruise / Famous Austin ('Austinpussy')
          )
        </p>
      </li>
      <li>
        <p>
          Knight and Day
           (
          Roy Miller
          )
        </p>
      </li>
      <li>
        <p>
          Mission: Impossible - Fallout
           (
          Ethan Hunt
          )
        </p>
      </li>
      <li>
        <p>
          Valkyrie
           (
          Claus Schenk Graf von Stauffenberg
          )
        </p>
      </li>
    </ul>
  </section>
</article>
`;
